src/nbl/video/COpenGLCommandPool.h

```diff
@@ -39,7 +39,13 @@ class COpenGLCommandPool final : public IGPUCommandPool
             std::unique_lock<std::mutex> lk(mutex);
             mempool.free_n<T>(const_cast<T*>(ptr), n);
         }
-        
+
+        void reset() override
+        {
+            // TODO reset some memory here?
```

Comment:
It should act as if all the command buffer allocated from this pool were reset with `E_RESET_FLAGS::ERF_RELEASE_RESOURCES_BIT`

Reply:
the reasoning is that whenever `checkForCommandPoolReset` returns true, it means all the data storage for the command "tapes" is already gone, so you'd need to do
```cpp
inline void IGPUCommandPool::reset()
{
   m_state = INITIAL;
   // all memory that backs references already gone and destructed
   if (checkForCommandPoolReset())
      return;

   // invalid usage
   if (!canReset())
   {
      NBL_DEBUG_BREAK(false);
      m_state = INVALID;
   }

   // TODO: iterate over the tapes and free their allocations
   reset_impl();
}
```

Reply:
Note: This is most likely `IGPUCommandBuffer::reset`.

Reply:
I also want to point out that transitioning the state of a non-resettable command buffer to `INVALID` because it fails the `reset` call, is not consistent with the Vulkan spec.

Reply:
Resolved.